server/controllers/profile.js

```diff
@@ -6,119 +6,143 @@ const generateToken = require("../utils/generateToken");
 // @desc Given parameters passed in create a profile
 // @access Private
 
- exports.createProfile = asyncHandler(async (req, res, next) => {
-
-    const { firstName, lastName, description, email, phoneNumber, availability:{startDate,endDate,daysOfWeek} } = req.query;
-    
-    const profile = await Profile.create({
-        firstName,
-        lastName,
-        description,
-        email,
-        phoneNumber,
-        availability:{
-          startDate,
-          endDate,
-          daysOfWeek: []
-      }
-      });
-    
-      if (profile) {
-        const token = generateToken(profile._id);
-        const secondsInWeek = 604800;
-    
-        res.cookie("token", token, {
-          httpOnly: true,
-          maxAge: secondsInWeek * 1000
-        });
-    
-        res.status(201).json({
-          success: {
-            user: {
-              id: profile._id,
-              ffirstName: profile.firstName,
-              lastName: profile.lastName,
-              description: profile.description,
-              email: profile.email,
-              phoneNumber: profile.phoneNumber,
-              availability:{
-                startDate: profile.availability.startDate,
-                endDate: profile.availability.endDate,
-                daysOfWeek: []
-            }
-            }
-          }
-        });
-      } else {
-        res.status(400);
-        throw new Error("Invalid user data");
-      }
-})
- 
+exports.createProfile = asyncHandler(async (req, res, next) => {
+  console.log(req.body);
+  const new_obj = Object.assign(
+    {
+      availability: {
+        startDate: "2012-04-23T18:25:43.511Z",
+        endDate: "2012-04-23T18:25:43.511Z",
+        daysOfWeek: ["Saturday", "Monday"],
+      },
+    },
+    req.body
+  );
+  console.log(new_obj);
+  const {
+    firstName,
+    lastName,
+    description,
+    email,
+    phoneNumber,
+    availability: { startDate, endDate, daysOfWeek },
+  } = new_obj;
+
+  const profile = await Profile.create({
+    firstName,
+    lastName,
+    description,
+    email,
+    phoneNumber,
+    availability: {
+      startDate,
+      endDate,
+      daysOfWeek,
+    },
+  });
+
+  if (profile) {
+    const token = generateToken(profile._id);
+    const secondsInWeek = 604800;
+
+    res.cookie("token", token, {
+      httpOnly: true,
+      maxAge: secondsInWeek * 1000,
+    });
+
+    res.status(201).json({
+      success: {
+        user: {
+          id: profile._id,
+          ffirstName: profile.firstName,
+          lastName: profile.lastName,
+          description: profile.description,
+          email: profile.email,
+          phoneNumber: profile.phoneNumber,
+          availability: {
+            startDate: profile.availability.startDate,
+            endDate: profile.availability.endDate,
+            daysOfWeek: [],
```

Comment:
Shouldn't we send `profile.availability.daysOfWeek` here?

Reply:
it's under TODO